plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/HeteroDescribableConfigurator.java

```diff
@@ -120,23 +121,13 @@ public CNode describe(T instance, ConfigurationContext context) {
     }
 
     @CheckForNull
-    public Map<String,List<String>> describeStructure(T instance, ConfigurationContext context) {
-        Map<String, List<String>> heteroConfigurationMapper = new LinkedHashMap<>();
-        lookupConfigurator(context, instance.getClass())
-            .map(configurator -> convertToNode(context, configurator, instance))
-            .filter(Objects::nonNull)
-            .map(node -> {
-
-                //Could something like this be done
-                    String BaseConfiguratorNode =  node.getClass().getName();
-                    List<String> propertiesList = new ArrayList<>();
-                    for(CNode cNode: node) {
-                        propertiesList.add(cNode.toString());
-                    }
-                 return heteroConfigurationMapper;
-            }).getOrNull();
-
-        return null;
+    public CNode describeStructure(T instance, ConfigurationContext context) {
+        return lookupConfigurator(context, instance.getClass())
+                            .map(configurator -> convertToNode(context, configurator, instance))
+                            .filter(Objects::nonNull)
+                            .map(node->{
+                              return new Scalar(preferredSymbol(instance.getDescriptor()));
+                            }).getOrNull();
```

Comment:
@casz This implementation would return all the CNodes am I right, I cannot test it currently so need to rely on feedback. I might have not understood what you meant a bit clearly but would we want to  use `describeStructure` to return the CNode of all the configurators.Is this a bit close to what we want? Would this CNode have all the info we need?

Reply:
CNode is quite confusing to me :cry:

Reply:
CNode is just a object that is meant to separate the mapping away from YAML so if someone wanted we could add support for another format than YAML.

Reply:
So the above would return all the base configurators for the heteroDescribableConfigurators ?